controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -278,6 +244,58 @@ public void handleResult(OperationContext.ResultAction resultAction, OperationCo
          * @throws OperationFailedException
          */
         abstract void doExecute(OperationContext context, ModelNode operation, FilteredData filteredData, boolean ignoreMissingResource) throws OperationFailedException;
+
+        private static class MultiTargetResultHandler implements OperationContext.ResultHandler {
+
+            private final FilteredData localFilteredData;
+            private final ModelNode result;
+            private final ModelNode fakeOperationResponse;
+
+            public MultiTargetResultHandler(ModelNode fakeOperationResponse, FilteredData localFilteredData, ModelNode result) {
+                this.localFilteredData = localFilteredData;
+                this.result = result;
+                this.fakeOperationResponse = fakeOperationResponse;
+            }
+
+            @Override
+            public void handleResult(OperationContext.ResultAction resultAction, OperationContext context, ModelNode operation) {
+                if(fakeOperationResponse != null && fakeOperationResponse.hasDefined(FAILURE_DESCRIPTION)) {
+                    context.setRollbackOnly();
```

Comment:
Is this necessary? I don't think it should be needed, since whatever step populated fakeOperationResponse has already reported a problem. So I think it's out-of-scope for this result handler to be manipulating the OC's resultAction.